analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs

```diff
@@ -28,9 +28,12 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn
 {
     public sealed record SymbolicValue
     {
+        // Reuse instances to save memory. This "True" has the same semantic meaning and any other symbolic value with BoolConstraint.True constraint
```

Comment:
At the moment, there are only a couple of distinct SymbolicValue possible. We could pool all of these.

Reply:
It will get more tricky with more constraint from other rules